.github/workflows/unitpacker-go.yml

```
@@ -0,0 +1,255 @@
+# * Your Apple Developer ID Installer Certificate
+# Base64-encode the resulting p12 file and paste the string in this secret.
+# The Installer signing identity is used to sign packages.
+###
+# MAC_INST_CERTIFICATE_PASSPHRASE - passphrase for the MAC_INST_CERTIFICATE private key
+# The passphrase that you used to encrypt your private key during the export.
+###
+# MAC_INST_CERTIFICATE_CN - common name on the MAC_INST_CERTIFICATE
+# This looks something like this: "Developer ID Installer: <Your name or company name> (<TeamID>)"
+# If it does not start with "Developer ID Installer: " it is the incorrect certificate or name.
+###
+###
+# MAC_STORE_APPLE_ID - Apple ID of the user requesting notarization
+# Usually an e-mail address.
+###
+# MAC_STORE_APP_SPECIFIC_PASSWORD - Application specific password of the user requesting notarization.
+# You can request an app-specific password for your Apple ID user at https://appleid.apple.com
+###
+# MAC_STORE_TEAM_ID - Team ID of the user or organization
+# This is the same 10 characters that are displayed in your certificate CN at the end in brackets.
+###
+# GPG_PRIVATE_KEY - PGP private key block for Linux signing
+# GPG_PASSPHRASE - passphrase for the GPG_PRIVATE_KEY
+##
+
+# https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows
+# > Note: An event will not be created when you push more than three tags at once.
+
+name: unitpacker-go
+on:
+  push:
+    tags:
+      - "multisig-v*"
+
+jobs:
+
+  config:
+    runs-on: ubuntu-latest
+    outputs:
+      config: ${{ steps.set-config.outputs.config }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: set-config
+        run: |
+          if [ -n "${ACT+x}" ]; then
+            ./compile-unitpacker-config.sh gaiad-v99.0.1
+          else
+            ./compile-unitpacker-config-tools.sh "$GITHUB_REF_NAME"
+          fi
+
+  config-test:
+    needs: config
+    strategy:
+      matrix: ${{fromJSON(needs.config.outputs.config)}}
+    runs-on: ${{matrix.os}}
+    steps:
+      - run: |
+          echo "hello world"
+          echo "Tool name: ${{matrix.tools.name}}"
+          echo "Version: ${{matrix.tools.ref}}"
+          echo "Git repo: ${{matrix.tools.git_url}}"
+
+  build:
+    needs: config
+    strategy:
+      matrix: ${{ fromJSON(needs.config.outputs.config) }}
+    runs-on: ${{matrix.os}}
+    steps:
+
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ matrix.tools.git_url }}
+          ref: ${{ matrix.tools.ref }}
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '>=1.19.0'
+
+      - name: Install binary
+        run: go install
+
+      - name: Print binary version
+        run: ~/go/bin/${{ matrix.tools.name }} --version
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ github.run_id }}.${{ matrix.os }}
+          if-no-files-found: error
+          retention-days: 3
+          path: ~/go/bin/${{ matrix.tools.name }}
```

For future proofing, I would enable uploading Windows-compatible binaries here. (Windows requires the full name to upload, which is `multisig.exe` in our case`.)

By the way, did you try to build this for Windows? I think it should be doable. Although without the network binaries, it's less useful. :(

```suggestion
          path: ~/go/bin/${{ matrix.tools.name }}*
```

---

Agree with this. Updated the code.